week-2-wide-perspective/w02e01-functional-warm-up/src/getDataFunctions.js

```diff
@@ -0,0 +1,33 @@
+export function getItemByKey(obj, key) {
+  return obj[key];
+}
+
+export function getAllItemsByKey(array, key) {
+  return array.map(item => item[key]);
```

Comment:
Tutaj przydała by się kompozycja. Zobacz @teziovsky. chwilę wyżej masz zapisane coś takiego: `getItemByKey`

nie fajniej z tego skorzystać ?

```typescript
return array.map(item => getItemByKey(item, key));
```
jakbyś wrzucił to (`getItemByKey`) w _[closure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures)_ albo zrobił _[curry](https://javascript.info/currying-partials)_ to wtedy moglibyśmy zapisać tak:

```typescript
return array.map(getItemByKey(key));
```

Reply:
No fakt, przeoczyłem tutaj wykorzystanie funkcji, którą wcześniej zaimplementowałem 🤔